libsignal-service/src/session_store.rs

```diff
@@ -26,4 +28,42 @@ pub trait SessionStoreExt: SessionStore {
         &self,
         address: &str,
     ) -> Result<usize, SignalProtocolError>;
+
+    /// Remove a session record for a recipient ID + device ID tuple.
+    async fn delete_service_addr_device_session(
```

Comment:
The implementation looks good to me, but I have the following questions:

1. since we probably don't want users to implement this method themselves, what about having it as a method of `ServiceAddress` instead? We already have `get_sub_device_sessions` there.
2. I also not sure if we should use `ServiceAddress` in those traits, since we might want to try and merge them back to `libsignal-protocol`?

Reply:
> I also not sure if we should use `ServiceAddress` in those traits, since we might want to try and merge them back to `libsignal-protocol`?

You would rather pass `uuid`, `e164` and `device_id` separately? IMO, this is a feature of the session store, libsignal-protocol only cares about one session at a time.

> since we probably don't want users to implement this method themselves, what about having it as a method of `ServiceAddress` instead? We already have `get_sub_device_sessions` there.

Hmm, I get why you say to keep this method non-overridable; maybe it shouldn't be part of the public API indeed. OTOH, moving it into ServiceAddress would mean that you'd call `service_address.delete_sessions_for_device(&mut some_session_store, device_id)`, which IMO sounds even worse.

What I could maybe get behind, is a `pub(crate) trait SessionStoreExtPrivate: SessionStore {}` which implements these kind of methods.

Reply:
Argument *pro* having this method overridable: an SQL back-end could implement this as in `DELETE FROM sessions WHERE uuid={} OR e164={}`.

I think we should just merge as-is, no?

Reply:
Sure, go ahead!